synapse/handlers/room_member.py

```diff
@@ -780,6 +781,20 @@ def do_3pid_invite(
                 txn_id=txn_id
             )
 
+    def _get_id_server_target(self, id_server):
+        """Looks up an id_server's actual http endpoint
+
+        Args:
+            id_server (str): the server name to lookup.
+
+        Returns:
+            the http endpoint to connect to.
+        """
+        if id_server in self.rewrite_identity_server_urls:
+            return self.rewrite_identity_server_urls[id_server]
```

Comment:
(fwiw you could write `return self.rewrite_identity_server_urls.get(id_server, id_server)`)

Reply:
oh, that's neat